components/scream/src/physics/p3/atmosphere_microphysics.cpp

```diff
@@ -94,6 +94,79 @@ void P3Microphysics::set_grids(const std::shared_ptr<const GridsManager> grids_m
 
 }
 
+// =========================================================================================
+int P3Microphysics::requested_buffer_size_in_bytes() const
+{
+  const Int nk_pack = ekat::npack<Spack>(m_num_levs);
+
+  // Number of Reals needed by local views in the interface
+  const int interface_request = Buffer::num_1d_scalar*m_num_cols*sizeof(Real) +
+                                Buffer::num_2d_vector*m_num_cols*nk_pack*sizeof(Spack) +
+                                Buffer::num_2d_scalar*m_num_cols*3*sizeof(Real);
```

Comment:
It seems 1d are (ncols) quantities, and 2d  are (ncols,nlevs). And scalar vs vector is to say non-packed vs packed? But then, why are 2d scalars of size ncols*3?

Reply:
Yeah, this was a random local variable called `col_location`. It's the only 2d local scalar view. I'll change this to not be under the generic category `num_2d_scalar`.